tvb_library/tvb/datasets/base.py

```diff
@@ -0,0 +1,19 @@
+
+
+class ZenodoDataset:
+
+    def __init__(self, version):
+        self.version = version
```

Comment:
```
from tvb.basic.logger.builder import get_logger
from tvb.basic.profile import TvbProfile

class BaseDataset:

    def __init__(self, version, target_download=None):

        self.log = get_logger(self.__class__.__module__)
        self.cached_files = None
        self.version = version

        if target_download is None:
            target_download = TvbProfile.current.DATASETS_FOLDER
        if os.path.exists(target_download):
            self.log(f"Folder {target_download} already exists, it will not be created again.")
        else:
            self.log(f"Folder {target_download} does not exist, we will create it now ...")
            os.mkdir(target_download)
        self.target_download = target_download
```
And in `tvb.basic.config.profile_settings.py` BaseSettingsProfile have a new attribute initialised in init: `self.DATASETS_FOLDER = os.path.join(self.TVB_STORAGE, "DATASETS")`

Reply:
this looks good!

Reply:
i feel that the `target_download` is not required when we are instantiating TVBZenodoDataset class but when we are fetching the data.
```python
        if target_download is None:
            target_download = TvbProfile.current.DATASETS_FOLDER
        if os.path.exists(target_download):
            self.log(f"Folder {target_download} already exists, it will not be created again.")
        else:
            self.log(f"Folder {target_download} does not exist, we will create it now ...")
            os.mkdir(target_download)
         
```
So, above part looks good to me, but it should be placed in fetch_data method. 

```python 

          self.target_download = target_download  
```

Im bit against of linking the target_download variable to the class because this way we are fixing the directory where the data will be extracted. 

Lets say one wants to extract `connectivity_66.zip` to `dir1` and `connectivity_192.zip` to `dir2`, he will have to create two separate instances. Keeping it as a parameter to `fetch_data` method would be better in my opinion.   

```python 
#when target_download is used while instantiating class. 

connectivity_66 = TVBZenodoDataset(target_download = "dir1").fetch_data("connectivity_66.zip")
connectivity_192 = TVBZenodoDataset(target_download = "dir1").fetch_data("connectivity_192.zip")

#we needed two separate instances of the class 
```

vs 

```python
#when target_download is a parameter for fetch_data class

tvb_data = TVBZenodoDataset()
connectivity_66 = tvb_data.fetch_data("connectivity_66.zip", "dir1")
connectivity_192 =tvb_data.fetch_data("connectivity_192.zip", "dir2")

#only one instance is enough.
```

Let me know what you think @liadomide .

Reply:
In my opinion one single `target_download` per instance of `BaseDataset ` is enough.
I feel that changing the `target_download` is happening rarely.
If we keep it in the  `fetch_data` as param, then the user needs to pass that folder path multiple times (which might be annoying, if the same value).

Reply:
okay, sounds good!